### ISSUE TITLE
Edge backend leaks processes

### ISSUE THREAD

**Issue by dkaste** [Boscop/web-view#74]
When running a WebView with the Edge backend, two "Desktop App Web Viewer" processes are started. If the WebView window is closed (clicking the "x" or alt+f4), these processes are terminated. However, if `WebView::terminate()` is called instead, these processes are never ended and will outlive the main process.

I messed around a bit with the C++ implementation of the Edge backend, and I could only get one of the processes to close (via `m_process.Terminate()`).

**Comment by zxey** [comment]
Could you post what have you changed in the c++ implementation? Also what did you you use for monitoring processes? I used Sysinternals Process Explorer, and I can see three processes being created: Win32WebViewHost.exe, WWAHost.exe and RuntimeBroker.exe, the first two get terminated properly, but RuntimeBroker.exe leaks, is this the same thing you are experiencing?

**Comment by dkaste** [comment]
I just used the default Windows Task Manager to monitor processes.

Using the details tab, the processes created are like you said:
* `WWAHost.exe`
* `Win32WebViewHost.exe`
* `RuntimeBroker.exe`

I hadn't initially noticed `RuntimeBroker.exe`, but it is created as well. However, when I exit my application with `WebView::terminate()`, none of these processes end. If I exit my application by closing the actual window, only RuntimeBroker survives.

As for the C++ changes, I just overrode `browser_window::terminate()` in the `webview` class like so:
```c++
void terminate() {
    browser_window::terminate();
    m_process.Terminate();
}
```
This change only terminates the `WWAHost.exe` process.

Another unfortunate thing to note is that if the application is closed by Ctrl+C, none of the processes get killed either.

**Comment by dkaste** [comment]
Perhaps [Job Objects](https://docs.microsoft.com/en-us/windows/win32/procthread/job-objects) could be used for a robust solution here?

**Comment by zxey** [comment]
Yes, it could be. But first we need to find why this happens. It seems that `WebViewControlProcess` creates  one process and `CreateWebViewControlAsync` creates another.

**Comment by dkaste** [comment]
I have opened a pull request to fix the main problem here, but I will leave this issue open. The remaining problem is that if an application is closed ungracefully (Ctrl+C, "End task", etc.), processes will still be leaked.

**Comment by zxey** [comment]
For Ctrl+c we will probably have to catch Windows equivalent of SIGINT so we can do graceful shutdown. Not sure if some signal can be caught for "End Task", will need to investigate.

**Comment by dkaste** [comment]
More information: windows-toolkit/Microsoft.Toolkit.Win32#146

It's possible that this issue will eventually be resolved in Windows itself.
